### PR TITLE
feat(server): added support for SVG files

### PR DESCRIPTION
**Pull request by declan8010**
This PR enhances the Immich open-source project by adding support for Scalable Vector Graphics (SVG) files. With this feature, users can now upload SVG files as assets and view them within the application. SVG files offer several advantages, including scalability and resolution independence, making them a valuable addition to the supported asset formats.

While I have thoroughly tested this implementation on the web interface, the integration with mobile clients remains untested. I encourage our contributors to review, expand, and refine this feature as needed.

There is going to be some rough edges so I invite any of our contributers to add or remove anything that will help 👍 

image

### CODE REVIEW

File: server/src/immich/api-v1/asset/asset.service.ts [hunk 1]
```diff
@@ -338,7 +340,7 @@ export class AssetService {
       fileModifiedAt: dto.fileModifiedAt,
       localDateTime: dto.fileCreatedAt,
 
-      type: mimeTypes.assetType(file.originalPath),
```
Comment (alextran1502) [inline server/src/immich/api-v1/asset/asset.service.ts:341]: We can leave this as is, it is self-explanatory so you don't need to add a comment

**Review by alextran1502 — APPROVED** [review]

File: server/src/immich/api-v1/asset/asset.service.ts (hunk 1 above)
Comment (declan8010) [reply server/src/immich/api-v1/asset/asset.service.ts:341]: Actioned

**Comment by alextran1502** [comment]
I mean you can keep this `type: mimeTypes.assetType(file.originalPath),` instead of creating a local variable for the method :)

**Comment by declan8010** [comment]
@alextran1502 Would you be able to take a look at why the server test is failing if you have a moment?

**Comment by alextran1502** [comment]
You can read about why it failed here https://github.com/immich-app/immich/actions/runs/8286816641/job/22677716034?pr=7960

**Comment by declan8010** [comment]
> You can read about why it failed here https://github.com/immich-app/immich/actions/runs/8286816641/job/22677716034?pr=7960

Its strange that its still failing even after I moved the mimetype to be in the correct order

image

**Comment by jrasm91** [comment]
I think there are some security issues related to svgs that we previously discussed. cc: @mertalev

**Comment by alextran1502** [comment]
@jrasm91 I remember now, like add code to SVG file?

**Comment by jrasm91** [comment]
SVGs can container javascript and they'll be executed, etc. by the browser. There are other security considerations as well I believe, like SVGs can have references to external scripts, etc.

**Comment by mertalev** [comment]
Yes, there are some [security issues](https://medium.com/sse-blog/beware-svg-its-not-just-a-scalable-graphic-9a2e9f3df6ef) with user-provided SVG. They're safe if used in `img` blocks, but if the user right-clicks the image and opens it in another tab they can have arbitrary code executed.

**Comment by alextran1502** [comment]
Maybe we can make an option for the admin to disable uploading SVG files to the server later. So this change can be accepted now

**Pull request merged**